File: typescript/geometryUtils.ts
export class GeometryUtils {
  static calculateCircleArea(radius: number): number {
    return Math.PI * radius * radius;
  }
  
  static calculateCirclePerimeter(radius: number): number {
    return 2 * Math.PI * radius;
  }
  
  static calculateRectangleArea(length: number, width: number): number {
    return length * width;
  }
  
  static calculateRectanglePerimeter(length: number, width: number): number {
    return 2 * (length + width);
  }
  
  static calculateTriangleArea(base: number, height: number): number {
    return 0.5 * base * height;
  }
  
  static calculateDistance(x1: number, y1: number, x2: number, y2: number): number {
    return Math.sqrt(Math.pow(x2 - x1, 2) + Math.pow(y2 - y1, 2));
  }
  
  static calculateSphereVolume(radius: number): number {
    return (4/3) * Math.PI * Math.pow(radius, 3);
  }
  
  static calculateCylinderVolume(radius: number, height: number): number {
    return Math.PI * Math.pow(radius, 2) * height;
  }
  
  static calculateTriangleAreaWithSides(a: number, b: number, c: number): number {
    const s = (a + b + c) / 2;
    return Math.sqrt(s * (s - a) * (s - b) * (s - c));
  }
}

File: kotlin/GeometryUtils.kt
package kotlin

import kotlin.math.PI
import kotlin.math.pow
import kotlin.math.sqrt

object GeometryUtils {
    fun calculateCircleArea(radius: Double): Double = PI * radius * radius
    
    fun calculateCirclePerimeter(radius: Double): Double = 2 * PI * radius
    
    fun calculateRectangleArea(length: Double, width: Double): Double = length * width
    
    fun calculateRectanglePerimeter(length: Double, width: Double): Double = 2 * (length + width)
    
    fun calculateTriangleArea(base: Double, height: Double): Double = 0.5 * base * height
    
    fun calculateDistance(x1: Double, y1: Double, x2: Double, y2: Double): Double = 
        sqrt((x2 - x1).pow(2) + (y2 - y1).pow(2))
    
    fun calculateSphereVolume(radius: Double): Double = (4.0 / 3.0) * PI * radius.pow(3)
    
    fun calculateCylinderVolume(radius: Double, height: Double): Double = PI * radius.pow(2) * height
}